netket/hilbert/doubled_hilbert.py

```diff
@@ -0,0 +1,121 @@
+from .abstract_hilbert import AbstractHilbert
+from ..graph import DoubledGraph as _DoubledGraph
+
+import numpy as _np
+from netket import random as _random
+
+
+class PyDoubledHilbert(AbstractHilbert):
+    r"""Superoperatorial hilbert space for states living in the
+    tensorised state H\otimes H, encoded according to Choi's isomorphism."""
+
+    def __init__(self, hilb):
+        """Superoperatorial hilbert space for states living in the
+           tensorised state H\otimes H, encoded according to Choi's isomorphism.
+
+        Args:
+            hilb: the hilbrt space H.
+
+        Examples:
+            Simple superoperatorial hilbert space for few spins.
+
+           >>> from netket.graph import Hypercube
+           >>> from netket.hilbert import Spin, DoubledHilbert
+           >>> g = Hypercube(length=5,n_dim=2,pbc=True)
+           >>> hi = Spin(graph=g, s=0.5)
+           >>> hi2 = DoubledHilbert(hi)
+           >>> print(hi2.size)
+           50
+        """
+        doubled_graph = _DoubledGraph(hilb.graph)
+
+        self.graph = doubled_graph
+        self._size = doubled_graph.size
+        self.physical = hilb
+        self._hilbert_index = None
```

Comment:
not sure you need to define hilbert_index here since it's never used in this class